src/components/Sidebar.tsx

```diff
@@ -262,54 +264,54 @@ const Sidebar = ({ onClose, ...rest }: SidebarProps) => {
 
 export default Sidebar;
 
-function DropdownMenu({ title, icon, menuItems, rolesRequired } : {
+function DropdownMenu({ title, icon, menuItems, onClose } : {
   title: string,
   icon: React.ReactNode,
   menuItems: DropdownMenuItem[],
-  rolesRequired?: Role | Role[],
-}) {
+} & SidebarProps) {
   const [user] = useUserContext();
   const backgroundColor = useColorModeValue(bgColorModeValues[0], 
     bgColorModeValues[1]);
-  const borderColor = useColorModeValue(bdColorModeValues[0], 
-    bdColorModeValues[1]);
-  const filteredItems = rolesRequired ? 
-    managerDropdownMenuItems.filter(item => isPermitted(user.roles, item.roles))
-    :
-    menuItems;
-
+  const borderColor = useColorModeValue(borderColorModeValues[0], 
+    borderColorModeValues[1]);
+  const filteredItems = menuItems.filter(item => 
+    isPermitted(user.roles, item.roles));
+  
   if (filteredItems.length === 0) {
     return <></>;
   }
   return <Flex paddingY={sidebarItemPaddingY}>
     <Menu placement='right-start'>
     <DropdownMenuButton title={title} icon={icon}/>
     <MenuList bg={backgroundColor} borderColor={borderColor}>
-      {filteredItems.map((item, index) =>
-        <MenuItem 
-          key={index}
-          as={typeof item.action === 'string' ? NextLink : undefined}
-          href={typeof item.action === 'string' ? item.action : undefined}
-          onClick={typeof item.action === 'function' ? item.action : undefined} >
-          {item.icon}{item.name}
-        </MenuItem>)}
+      {filteredItems.map((item, index) => {
+        const isUrl = typeof item.action === 'string';
+        return (
+          <MenuItem 
+            key={index} 
+            // Only sets the link it is a url 
+            {...(isUrl && { as: NextLink })}
+            {...(isUrl && { href: item.action })}
+            onClick={() => {
+              if (typeof item.action === 'function') item.action();
```

Comment:
`if (!usUrl) ...`

Reply:
This will not pass the type checking. It may still consider it as as string, we cannot put a () after the string.

Reply:
can you use `(item as function).action()`?

Reply:
` (item as { action: Function }).action();` works